### PR TITLE
chore: Use `Path.getOrdersByType` in `calculateQuoteInfo` [LIT-690]

### PR DESCRIPTION
**Pull request by kyu-c**
# Description

* Use `Path.getOrdersByType` in `calculateQuoteInfo`

# Testing & Simbot Run

* Updated unit tests.

# Checklist

-   [x] Update 0x API documentation (gitbook) if needed (e.g. public facing API change).
-   [x] Add tests to cover changes as needed.
-   [x] All dependent changes (e.g. RFQ server, Gas Price Oracle) have been deployed (if any).

### CODE REVIEW

File: src/asset-swapper/utils/quote_info.ts [hunk 1]
```diff
@@ -27,14 +27,16 @@ export function calculateQuoteInfo(params: {
     slippage: number;
 }): QuoteInfo {
     const { path, operation, assetFillAmount, gasPrice, gasSchedule, slippage } = params;
+    const { nativeOrders, bridgeOrders, twoHopOrders } = path.getOrdersByType();
+    const singleHopOrders = [...nativeOrders, ...bridgeOrders];
     // TODO: generalize calculateQuoteInfo to handle a mix multihop+multiplex.
 
     // NOTES: until multihop+multiplex is supported a path will have single hop order(s) xor a two hop order.
```
Comment (kliao) [inline src/asset-swapper/utils/quote_info.ts:34]: "xor" or "or" ?

File: src/asset-swapper/utils/quote_info.ts [hunk 2]
```diff
@@ -27,14 +27,16 @@ export function calculateQuoteInfo(params: {
...
     if (path.hasTwoHop()) {
-        return calculateTwoHopQuoteInfo(path.getOrders(), operation, gasSchedule, slippage);
+        return calculateTwoHopQuoteInfo(twoHopOrders[0], operation, gasSchedule, slippage);
```
Comment (kliao) [inline src/asset-swapper/utils/quote_info.ts:36]: are we guaranteed only 1 twoHopOrder? a little confused by the use of the 0th index

File: test/asset-swapper/exchange_proxy_swap_quote_consumer_test.ts [hunk 3]
```diff
@@ -120,10 +120,11 @@ describe('ExchangeProxySwapQuoteConsumer', () => {
             makerToken: MAKER_TOKEN,
             takerToken: TAKER_TOKEN,
             path: {
+                getOrdersByType: () => ({ nativeOrders: [order], twoHopOrders: [], bridgeOrders: [] }),
                 getOrders: () => [order],
                 getSlippedOrders: (_maxSlippage: number) => [order],
                 hasTwoHop: () => false,
-            } as IPath,
```
Comment (kliao) [inline test/asset-swapper/exchange_proxy_swap_quote_consumer_test.ts:126]: can remove the IPath import

**Review by kliao — APPROVED** [review]

File: src/asset-swapper/utils/quote_info.ts (hunk 1 above)
Comment (kyu-c) [reply src/asset-swapper/utils/quote_info.ts:34]: exclusive or ( i guess all `or` is exclusive or in day-to-day language 🤣 )

(can't be both)

**Pull request merged**